aten/src/ATen/Context.cpp

```diff
@@ -59,6 +59,48 @@ void Context::setDeterministicCuDNN(bool b) {
   deterministic_cudnn = b;
 }
 
+bool Context::deterministic() const {
+  return _deterministic;
+}
+
+void Context::setDeterministic(bool b) {
+  _deterministic = b;
+}
+
+Context::ErrorLevel Context::deterministicErrorLevel() const {
+  return deterministic_error_level;
+}
+
+void Context::setDeterministicErrorLevel(Context::ErrorLevel e) {
+  deterministic_error_level = e;
+}
+
+Context::ErrorLevel Context::longToErrorLevel(long e) {
+  switch (e) {
+    case static_cast<int>(Context::ErrorLevel::None):
+    case static_cast<int>(Context::ErrorLevel::Warn):
+    case static_cast<int>(Context::ErrorLevel::Error):
+      return static_cast<Context::ErrorLevel>(e);
+    default:
+      TORCH_CHECK(false, "error level ", e, " is invalid, must be one of 0: None, 1: Warn, or 2: Error");
+  }
+}
+
+void Context::alertNotDeterministic(std::string const& caller) {
```

Comment:
nit: There isn't really any reason to take a const reference to a string here; if the user doesn't have a `std::string` hand all you're doing is forcing them to allocate one, just so that you can read it out. Pass a `const char*` or maybe a `c10::string_view` (if you're feeling fancy)